kolibri/core/auth/api.py

```diff
@@ -451,6 +458,25 @@ class FacilityUserViewSet(ValuesViewset):
         "is_superuser": lambda x: bool(x.pop("devicepermissions__is_superuser"))
     }
 
+    def destroy(self, request, *args, **kwargs):
+        if kwargs.get("pk"):
+            # Single object deletion
+            user = self.get_object()
+            if request.user.is_superuser and request.user.id == user.id:
+                return Response(status=status.HTTP_403_FORBIDDEN)
+            user.date_deleted = now()
+            user.save()
+            return Response(status=status.HTTP_204_NO_CONTENT)
+        else:
+            # Bulk deletion
+            return self.bulk_destroy(request, *args, **kwargs)
+
+    def perform_bulk_destroy(self, objects):
+        # Prevent deletion of superusers
+        if self.request.user.is_superuser:
+            objects = objects.exclude(id=self.request.user.id)
```

Comment:
Wouldn't the parallel behaviour for the bulk delete be more stringent? Something like:
```
if objects.filter(id=self.request.user.id).exists():
    raise PermissionDenied("Super user cannot delete self")
```